app/assets/javascripts/code/services/_fixed_sessions.js

```diff
@@ -116,8 +117,54 @@ export const fixedSessions = (
     },
 
     drawSessionsInLocation: function() {
-      map.markers = [];
-      (this.get()).forEach(session => drawSession.drawFixedSession(session));
+      if (sensors.anySelected() && this.scope.params.get('data').location.streaming) {
+        const sensorId = sensors.selectedId() || sensors.tmpSelectedId();
+        const sensor = sensors.sensors[sensorId] || {};
+        const selectedSensor = sensor.sensor_name;
+        (this.get()).forEach(session => this.drawColorCodedMarkers(session, selectedSensor));
+      } else {
+        (this.get()).forEach(session => this.drawDefaultMarkers(session));
+      }
+    },
+
+    drawColorCodedMarkers: function(session, selectedSensor) {
+      drawSession.undoDraw(session);
+      session.markers = [];
+
+      const content = Session.lastHourAverageVauleAndUnit(session, selectedSensor);
+      const heatLevel = heat.levelName(Session.lastHourAverage(session));
+      const latLng = Session.latLng(session);
+      const callback = (id) => () => $rootScope.$broadcast('markerSelected', {session_id: id});
+
+      if (!heat.outsideOfScope(heatLevel)) {
+        const customMarker = map.drawCustomMarker({
+            latLng: latLng,
+            content: content,
+            colorClass: heatLevel,
+            callback: callback(Session.id(session))
+          });
+        session.markers.push(customMarker);
+        map.markers.push(customMarker);
+        session.drawed = true;
+      }
+    },
+
+    drawDefaultMarkers: function(session) {
```

Comment:
Have you considered extracting the common code of `drawColorCodedMarkers` and `drawDefaultMarker` to another fnc?

Reply:
Good point. I'll do that but not just now. They look similar, but they refer to different concepts and I think they will diverge further with the next tickets, so I want to hold off on extracting this until I will know how the default markers logic will change. In the spirit of duplication is better than wrong abstraction :D